File: packages/server/UserService/Models/User/Comment.cs
using System.Text.Json.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace UserService.Models.User;

public class Comment
{
   [BsonId]
   [BsonRepresentation(BsonType.ObjectId)]
   [JsonPropertyName("_id")]
   public string Id { get; set; }
   [BsonElement("post")]
   [BsonRepresentation(BsonType.ObjectId)]
   public string PostId { get; set; }
   [BsonElement("commentText")]
   public string CommentText { get; set; }
   [BsonElement("createdAt")] 
   public DateTime CreatedAt { get; set; }
}

File: packages/server/CommentService/internal/core/domain/models/comment.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Comment struct {
	User        primitive.ObjectID `json:"user" bson:"user"`
	Post        primitive.ObjectID `json:"post" bson:"post"`
	CommentText string             `json:"commentText" bson:"commentText"`
	CreatedAt   time.Time          `json:"createdAt" bson:"createdAt"`
}
